shopinvader_api_address/routers/address_service.py

```diff
@@ -0,0 +1,143 @@
+from typing import Annotated, List
+
+from fastapi import APIRouter, Depends
+
+from odoo.addons.base.models.res_partner import Partner as ResPartner
+from odoo.addons.fastapi.dependencies import authenticated_partner
+from odoo.addons.shopinvader_schema_address.schemas import (
+    BillingAddress,
+    ShippingAddress,
+)
+
+from ..schemas import (
+    BillingAddressCreate,
+    BillingAddressUpdate,
+    ShippingAddressCreate,
+    ShippingAddressUpdate,
+)
+
+# create a router
+address_router = APIRouter(tags=["addresses"])
+
+# --- Billing addresses ---
+
+
+@address_router.get("/addresses/billing", response_model=List[BillingAddress])
+def get_billing_addresses(
+    partner: Annotated[ResPartner, Depends(authenticated_partner)],
+) -> List[BillingAddress]:
+    """
+    Get billing address of authenticated user
+    billing address corresponds to authenticated partner
+    """
+    address = partner._get_shopinvader_billing_addresses()
+    return [BillingAddress.from_res_partner(rec) for rec in address]
+
+
+@address_router.post(
+    "/addresses/billing", response_model=BillingAddress, status_code=201
+)
+def create_billing_address(
+    data: BillingAddressCreate,
+    partner: Annotated[ResPartner, Depends(authenticated_partner)],
+) -> BillingAddress:
+    """
+    Create billing address
+    Raise error since billing address is the authenticated partner
+    """
+    vals = data.to_res_partner_vals()
+    address = partner._create_shopinvader_billing_address(vals)
+    return BillingAddress.from_res_partner(address)
+
+
+@address_router.post("/addresses/billing/{address_id}", response_model=BillingAddress)
+def update_billing_address(
+    data: BillingAddressUpdate,
+    partner: Annotated[ResPartner, Depends(authenticated_partner)],
+    address_id: int,
+) -> BillingAddress:
+    """
+    Update billing address of authenticated user
+    billing address corresponds to authenticated partner
+    """
+    vals = data.to_res_partner_vals()
+    # sudo() is needed because some addons override the write
+    # function of res.partner to do some checks before writing.
+    # These checks need more rights than what we are giving to
+    # the enspoint's user
+    # (e.g. snailmail/models/res_partner.py)
+    address = partner.sudo()._update_shopinvader_billing_address(vals, address_id)
+    return BillingAddress.from_res_partner(address)
+
+
+# --- Shipping address ---
+
+
+@address_router.get("/addresses/shipping", response_model=List[ShippingAddress])
+@address_router.get(
+    "/addresses/shipping/{address_id}", response_model=List[ShippingAddress]
+)
```

Comment:
Does it have sense to keep /addresses/shipping/{address_id} ?
If yes, shouldn't we also have /addresses/**billing**/{address_id}?

Reply:
Yes I would keep it. And we need the same for the billing address indeed.
`/addresses/shipping/{address_id}` must return a single `ShippingAddress`, though, not a list.

Reply:
BTW with recent fastapi versions, response_model is not necessary anymore. The return type of the function is sufficient.

Reply:
No, I did it on the get of BillingAddress and ShippingAddress

Reply:
I finally have split the function in 2